src/library/Uno.Toolkit.Material/MaterialToolkitTheme.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Uno.Material;
+using Windows.UI;
+
+#if IS_WINUI
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Media;
+#else
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Media;
+#endif
+
+namespace Uno.Toolkit.UI.Material
+{
+	/// <summary>
+	/// Material (Material Design 3) styles for the controls in the Uno.Toolkit.UI library.
+	/// </summary>
+	public class MaterialToolkitTheme : ResourceDictionary
+	{
+		private bool _isOverridingColor;
+		private bool _isOverridingFont;
+
+#if IS_WINUI
+		private const string ToolkitPackageName = "Uno.Toolkit.WinUI";
+		private const string ToolkitMaterialPackageName = "Uno.Toolkit.WinUI.Material";
+		private const string PackageNameSuffix = "WinUI";
+#else
+		private const string ToolkitPackageName = "Uno.Toolkit.UI";
+		private const string ToolkitMaterialPackageName = "Uno.Toolkit.UI.Material";
+		private const string PackageNameSuffix = "UWP";
+#endif
+		#region FontOverrideSource (DP)
```

Comment:
just for consideration, lets use this for consistency?: https://github.com/Xiaoy312/bag-o-scripts/blob/main/DependencyProperty%20codegen.linq

Reply:
and move dp changed handler outside